source/deimos/openssl/opensslv.d

```diff
@@ -1,7 +1,99 @@
+/**
+ * Module to deal with the library version being used
+ *
+ * This library provide bindings for a wide range of OpenSSL versions,
+ * ranging from v0.9.x to v3.0.x. Some versions are not compatible with
+ * one another, either due to different ABI or different behavior,
+ * for example OpenSSL 1.0 requires initialization but later versions do not.
+ *
+ * While things tend to mostly work or error out while linking when the version
+ * the bindings assume and the actually C library version are too different,
+ * we prefer to try detecting the currently used version, and allow users
+ * to specify the version explicitly, before falling back to the latest bindings
+ */
 module deimos.openssl.opensslv;
 
 import deimos.openssl._d_util;
 
+version (DeimosOpenSSL_1_0_0)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-100
+    // OpenSSL 1.0.0t was released 2015-12-03
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"1.0.0t";
+}
+else version (DeimosOpenSSL_1_0_1)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-101
+    // OpenSSL 1.0.1u was released 2016-09-22
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"1.0.1u";
+}
+else version (DeimosOpenSSL_1_0_2)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-102
+    // OpenSSL 1.0.2t was released 2019-09-10
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"1.0.2t";
+}
+else version (DeimosOpenSSL_1_1_0)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-110
+    // OpenSSL 1.1.0l was released 2019-09-10
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"1.1.0l";
+}
+else version (DeimosOpenSSL_1_1_1)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-111
+    // OpenSSL 1.1.1m was released 2021-12-14
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"1.1.1m";
+}
+else version (DeimosOpenSSL_3_0)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-30
+    // OpenSSL 3.0.3 was released 2022-05-03
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"3.0.3";
+}
+else version (DeimosOpenSSL_3_1)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-31
+    // OpenSSL 3.0.3 was released 2022-05-03
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"3.0.3";
```

Comment:
```suggestion
     // OpenSSL 3.1.0 was released 2022-??-??
     public alias OpenSSLVersion = OpenSSLVersionTemplate!"3.1.0";
```
?

Actually I don't see it on https://github.com/openssl/openssl/tags .

Reply:
Yep, so why this addition?

https://github.com/D-Programming-Deimos/openssl/blob/7d8334f9d5b756d9c19a22cfbf9ee59ac4385ecf/source/deimos/openssl/opensslv.d#L54-L56

Reply:
My bad, their changelog mentions 3.1 so I originally added it, but couldn't actually find the release, so I decided to postpone it, but forgot to remove this branch.